### ISSUE TITLE
Use promises

### ISSUE THREAD

**Issue by veich** [lramsey/scikit-node#5]
This would look much better if implemented with promises.
I suggest bluebird http://bluebirdjs.com/docs/api/promise.coroutine.html

**Comment by lramsey** [comment]
Yeah, there is potential to clean this up in that way.  Thanks for the tip.
